Mathlib/Data/IsROrC/Basic.lean

```diff
@@ -143,236 +138,208 @@ theorem inv_def (z : K) : z⁻¹ = conj z * ((‖z‖ ^ 2)⁻¹ : ℝ) :=
-#align is_R_or_C.bit0_re IsROrC.bit0_re
-
-@[simp, is_R_or_C_simps]
-theorem bit1_re (z : K) : re (bit1 z) = bit1 (re z) := by
-  simp only [bit1, AddMonoidHom.map_add, bit0_re, add_right_inj, one_re]
-#align is_R_or_C.bit1_re IsROrC.bit1_re
-
-@[simp, is_R_or_C_simps]
-theorem bit0_im (z : K) : im (bit0 z) = bit0 (im z) := by simp only [bit0, map_add]
-#align is_R_or_C.bit0_im IsROrC.bit0_im
-
-@[simp, is_R_or_C_simps]
-theorem bit1_im (z : K) : im (bit1 z) = bit0 (im z) := by
-  simp only [bit1, add_right_eq_self, AddMonoidHom.map_add, bit0_im, one_im]
-#align is_R_or_C.bit1_im IsROrC.bit1_im
-
-theorem of_real_eq_zero {z : ℝ} : (z : K) = 0 ↔ z = 0 := by
-  rw [← of_real_zero] <;> exact of_real_inj
-#align is_R_or_C.of_real_eq_zero IsROrC.of_real_eq_zero
-
-theorem of_real_ne_zero {z : ℝ} : (z : K) ≠ 0 ↔ z ≠ 0 :=
-  of_real_eq_zero.Not
-#align is_R_or_C.of_real_ne_zero IsROrC.of_real_ne_zero
-
-@[simp, is_R_or_C_simps, norm_cast]
-theorem of_real_add ⦃r s : ℝ⦄ : ((r + s : ℝ) : K) = r + s := by
-  apply (@IsROrC.ext_iff K _ ((r + s : ℝ) : K) (r + s)).mpr
-  simp
-#align is_R_or_C.of_real_add IsROrC.of_real_add
-
-@[simp, is_R_or_C_simps, norm_cast]
-theorem of_real_bit0 (r : ℝ) : ((bit0 r : ℝ) : K) = bit0 (r : K) :=
-  ext_iff.2 <| by simp [bit0]
-#align is_R_or_C.of_real_bit0 IsROrC.of_real_bit0
-
-@[simp, is_R_or_C_simps, norm_cast]
-theorem of_real_bit1 (r : ℝ) : ((bit1 r : ℝ) : K) = bit1 (r : K) :=
-  ext_iff.2 <| by simp [bit1]
-#align is_R_or_C.of_real_bit1 IsROrC.of_real_bit1
-
-@[simp, norm_cast, is_R_or_C_simps]
-theorem of_real_neg (r : ℝ) : ((-r : ℝ) : K) = -r :=
-  ext_iff.2 <| by simp
-#align is_R_or_C.of_real_neg IsROrC.of_real_neg
-
-@[simp, norm_cast, is_R_or_C_simps]
-theorem of_real_mul (r s : ℝ) : ((r * s : ℝ) : K) = r * s :=
-  ext_iff.2 <| by simp [is_R_or_C_simps]
-#align is_R_or_C.of_real_mul IsROrC.of_real_mul
-
-@[simp, norm_cast, is_R_or_C_simps]
-theorem of_real_smul (r x : ℝ) : r • (x : K) = (r : K) * (x : K) := by
-  simp_rw [← smul_eq_mul, of_real_alg r]
-  simp only [Algebra.id.smul_eq_mul, one_mul, Algebra.smul_mul_assoc]
-#align is_R_or_C.of_real_smul IsROrC.of_real_smul
-
-@[is_R_or_C_simps]
-theorem of_real_mul_re (r : ℝ) (z : K) : re (↑r * z) = r * re z := by
-  simp only [mul_re, of_real_im, MulZeroClass.zero_mul, of_real_re, sub_zero]
-#align is_R_or_C.of_real_mul_re IsROrC.of_real_mul_re
-
-@[is_R_or_C_simps]
-theorem of_real_mul_im (r : ℝ) (z : K) : im (↑r * z) = r * im z := by
-  simp only [add_zero, of_real_im, MulZeroClass.zero_mul, of_real_re, mul_im]
-#align is_R_or_C.of_real_mul_im IsROrC.of_real_mul_im
-
-@[is_R_or_C_simps]
-theorem smul_re : ∀ (r : ℝ) (z : K), re (r • z) = r * re z := fun r z => by
-  rw [Algebra.smul_def]
-  apply of_real_mul_re
+theorem ofReal_inj {z w : ℝ} : (z : K) = (w : K) ↔ z = w := by simp [ext_iff]
+#align is_R_or_C.of_real_inj IsROrC.ofReal_inj
+
+#noalign is_R_or_C.bit0_re
+#noalign is_R_or_C.bit1_re
+#noalign is_R_or_C.bit0_im
+#noalign is_R_or_C.bit1_im
```

Comment:
I thought we were keeping these when portnig but marking them as deprecated

Reply:
What's the point of keeping them? They aren't useful for dealing with literal numerals anymore.

Reply:
Elsewhere in the port we've been keeping them, so it's probably better to be consistent and clear them all up at once.

Reply:
The porting wiki confirms that we should keep them.